find-closest-facility-from-point/src/main/java/com/esri/arcgismaps/sample/findclosestfacilityfrompoint/components/MapViewModel.kt

```diff
@@ -0,0 +1,199 @@
+     * to find the route between the incident and closest facility.
+     */
+    fun onSingleTapConfirmed(event: SingleTapConfirmedEvent) {
+
+        // Retrieve the tapped map point from the SingleTapConfirmedEvent
+        val mapPoint: Point = event.mapPoint ?: return messageDialogVM.showMessageDialog(
+            title = "No map point retrieved from tap.",
+            description = ""
+        )
+
+        // Create an incident on the tapped point
+        val incidentPoint = Incident(mapPoint)
+
+        // Clear and update the incident graphic on the map
+        updateIncidentGraphicPosition(incidentPoint)
+
+        // Find the closest facility to the incident
+        viewModelScope.launch {
+            findRoute(Incident(mapPoint))
+        }
+    }
+
+    /**
+     * Updates the [incidentGraphicsOverlay] to display a graphic on the [incidentPoint]
+     */
+    private fun updateIncidentGraphicPosition(incidentPoint: Incident) {
+
+        incidentGraphicsOverlay.graphics.clear()
+
+        // Create a cross symbol for the incident
+        val incidentMarker = SimpleMarkerSymbol(
+            style = SimpleMarkerSymbolStyle.Cross,
+            color = Color.black,
+            size = 20.0F
+        )
+
+        // Get the incident's geometry
+        val incidentStopGeometry = incidentPoint.geometry
+
+        // Add graphic to incident graphic overlays
+        incidentGraphicsOverlay.graphics.add(
+            Graphic(
+                geometry = incidentStopGeometry,
+                symbol = incidentMarker
+            )
+        )
+
+    }
+
+    /**
+     * Find the route between the given [incidentPoint] and the closest available facility
+     *
+     */
+    private suspend fun findRoute(incidentPoint: Incident) {
+        val closestFacilityTask = ClosestFacilityTask(
+            url = application.getString(R.string.san_diego_network_service_url)
+        )
+
+        // Create a job to find the route
+        val closestFacilityParameters: ClosestFacilityParameters =
+            closestFacilityTask.createDefaultParameters().getOrThrow()
+        // Add the facilities and incident points
+        closestFacilityParameters.apply {
+            setFacilities(facilitiesList)
+            setIncidents(listOf(incidentPoint))
+        }
+
+        // Solve a route using the facility task with the created parameters.
+        val closestFacilityResult =
+            closestFacilityTask.solveClosestFacility(closestFacilityParameters).getOrElse { error ->
+                messageDialogVM.showMessageDialog(
+                    title = "Error solving route: ${error.message.toString()}",
+                    description = error.cause.toString()
+                )
+            } as ClosestFacilityResult
+        val rankedFacilitiesList = closestFacilityResult.getRankedFacilityIndexes(incidentIndex = 0)
+        // If we got a list of facilities
```

Comment:
Let's explain the purpose of this if condition here: 

```suggestion
        // If the result contains a facility, solve the route from incident to facility. 
```